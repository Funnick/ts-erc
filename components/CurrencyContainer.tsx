import CurrencyColumn from './CurrencyColumn';

import { FlexContainer, IconColumn, Icon } from './styled/styledComponents';

type CurrencyContainerProps = {
    codes: { [currency: string]: string }
    fromCurrency: string
    setFromCurrency: (value: string) => void
    handleSwap: () => void
    toCurrency: string
    setToCurrency: (value: string) => void
    currencies: string[]
}

const CurrencyContainer = ({ codes, fromCurrency, setFromCurrency, handleSwap, toCurrency, setToCurrency, currencies }: CurrencyContainerProps) => {
  return (
        <FlexContainer>
            <CurrencyColumn title="From" flagCode={codes[fromCurrency]} currency={fromCurrency}
            setCurrency={setFromCurrency}  currencies={currencies} />

            <IconColumn role="icon-column">
                <Icon role="icon" className="fas fa-exchange-alt" onClick={handleSwap} />                
            </IconColumn>

            <CurrencyColumn title="To" flagCode={codes[toCurrency]} currency={toCurrency}
            setCurrency={setToCurrency}  currencies={currencies} />
        </FlexContainer>
)}

export default CurrencyContainer