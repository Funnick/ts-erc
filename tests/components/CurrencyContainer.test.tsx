import { render, screen } from '@testing-library/react';
import '@testing-library/jest-dom';
import renderer from 'react-test-renderer';
import CurrencyContainer from '@/components/CurrencyContainer';

describe('Component Test CurrencyContainer', () => {
    it('Should render properly', () => {
        render(<CurrencyContainer codes={{"EUR":"RF", "CAD":"CA", "USD":"US"}}
        fromCurrency={"EUR"} setFromCurrency={(s: string) => {}}
        handleSwap={() => {}} toCurrency={"USD"}
        setToCurrency={(s: string) => {}} currencies={["EUR", "CAD", "USD"]} />)
        
        const icon = screen.getByRole('icon')
        const iconColumn = screen.getByRole('icon-column')
        
        expect(icon).toBeInTheDocument()
        expect(iconColumn).toBeInTheDocument()
    })

    it('Snapshot', () => {
        const tree = renderer.create(<CurrencyContainer codes={{"EUR":"RF", "CAD":"CA", "USD":"US"}}
        fromCurrency={"EUR"} setFromCurrency={(s: string) => {}}
        handleSwap={() => {}} toCurrency={"USD"}
        setToCurrency={(s: string) => {}} currencies={["EUR", "CAD", "USD"]} />).toJSON()
        expect(tree).toMatchSnapshot()
    })
})

