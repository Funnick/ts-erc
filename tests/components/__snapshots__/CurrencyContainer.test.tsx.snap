// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component Test CurrencyContainer Snapshot 1`] = `
<div
  className="sc-kPTPQs dVARFL"
>
  <div
    className="sc-ispOId duzTmJ"
  >
    <p
      className="sc-fFGjHI fxeOjw"
    >
      From
    </p>
    <div
      className="sc-kcuKUB cgcjto"
    >
      <img
        alt="RF-flag"
        className="sc-dcrmVg deDNXi"
        height={25}
        src="https://flagsapi.com/RF/flat/64.png"
        width={25}
      />
      <select
        className="sc-kuWgmH dWoOPP"
        onChange={[Function]}
        value="EUR"
      >
        <option
          className="sc-duSInm"
          value="EUR"
        >
          EUR
        </option>
        <option
          className="sc-duSInm"
          value="CAD"
        >
          CAD
        </option>
        <option
          className="sc-duSInm"
          value="USD"
        >
          USD
        </option>
      </select>
    </div>
  </div>
  <div
    className="sc-ispOId sc-jrrXlR duzTmJ gjvJFm"
    role="icon-column"
  >
    <i
      className="sc-crozmw bDavlu fas fa-exchange-alt"
      onClick={[Function]}
      role="icon"
    />
  </div>
  <div
    className="sc-ispOId duzTmJ"
  >
    <p
      className="sc-fFGjHI fxeOjw"
    >
      To
    </p>
    <div
      className="sc-kcuKUB cgcjto"
    >
      <img
        alt="US-flag"
        className="sc-dcrmVg deDNXi"
        height={25}
        src="https://flagsapi.com/US/flat/64.png"
        width={25}
      />
      <select
        className="sc-kuWgmH dWoOPP"
        onChange={[Function]}
        value="USD"
      >
        <option
          className="sc-duSInm"
          value="EUR"
        >
          EUR
        </option>
        <option
          className="sc-duSInm"
          value="CAD"
        >
          CAD
        </option>
        <option
          className="sc-duSInm"
          value="USD"
        >
          USD
        </option>
      </select>
    </div>
  </div>
</div>
`;
